### PR TITLE
[Instrumentation.Runtime] Fix README description of the value for GC collections count

### PR DESCRIPTION
**Pull request by spanglerco**
When going through the README for [process.runtime.dotnet.gc.collections.count](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Instrumentation.Runtime#processruntimedotnetgccollectionscount), I was confused by the description for the value. I believe it's backwards: it's not that younger generations need to be excluded, but older generations ([the code](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/4b8e1d156b3f9eac2d6ebe99749f343889e67ffb/src/OpenTelemetry.Instrumentation.Runtime/RuntimeMetrics.cs#L237) iterates backwards and subtracts as it goes down).

I confirmed the metric values make sense when you consider gen1 to be `GC.CollectionCount(1) - GC.CollectionCount(2)` (and gen0 would subtract both 1 and 2).

## Changes

Trivial update to the README to indicate that the value for each generation excludes the counts from older generations instead of younger ones.

### CODE REVIEW

**Review by lachmatt — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

**Comment by Kielek** [comment]
@spanglerco, could you please sign CLA?

**Comment by spanglerco** [comment]
@Kielek Sorry for the delay, I think I technically need to sign the corporate CLA in case I make future non-trivial commits, and I'm trying to track down the right contact.

File: src/OpenTelemetry.Instrumentation.Runtime/README.md [hunk 1]
```diff
@@ -49,9 +49,9 @@ to the application.
 Number of garbage collections that have occurred since process start.
 
 Note: Collecting a generation means collecting objects in that generation and all
-its younger generations. However, each dimension for this metrics doesn't include
-the collection counts for the lower generation.
-e.g. count for gen1 is `GC.CollectionCount(1) - GC.CollectionCount(0)`.
+its younger generations. However, each dimension for this metric doesn't include
+the counts for collections triggered by higher generations.
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Runtime/README.md:53]:
```suggestion
the counts for collections for the higher generations.
```

Comment (cijothomas) [reply src/OpenTelemetry.Instrumentation.Runtime/README.md:53]: @noahfalk please review for accuracy.

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.Runtime/README.md [hunk 2]
```diff
@@ -49,9 +49,9 @@ to the application.
 Number of garbage collections that have occurred since process start.
 
 Note: Collecting a generation means collecting objects in that generation and all
```
[the remark this answers is not in the document]
Comment (spanglerco) [reply src/OpenTelemetry.Instrumentation.Runtime/README.md]: That's a great way to expand the description, thanks. I updated the PR.

**Review by cijothomas — APPROVED** [review]

**Pull request merged**